modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/resolveField.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {
+	FDS_ARRAY_FIELD_NAME_DELIMITER,
+	FDS_ARRAY_FIELD_NAME_PARENT_SUFFIX,
+	FDS_NESTED_FIELD_NAME_DELIMITER,
+	FDS_NESTED_FIELD_NAME_PARENT_SUFFIX,
+} from '../constants';
+
+export const resolveField = function resolveField(
+	path: string | Array<string>,
+	item: any
+) {
```

Comment:
I am really not a fan of utils that return objects with unrelated values or a tuple. It adds a layer of awkward naming, as you need to somehow name the frankenstein structure. This should be a `get*` of _something_, but nothing really sounds right.

Would it maybe make sense to move the content of `resolveField` inline into `getLocalizedValue`? There is only one invocation of it `resolveField`, and it's not exported. Then, we can rename `getLocalizedValue` into something like `getRendererProps`, that returns `IRendererProps` objects. Tests can be on that util, and include localization aspect. What do you think?

We don't need to do something like this in this PR, but it's seems like a good moment, at least to consider it.